Atomic/syscontainers.py

```diff
@@ -386,6 +386,147 @@ def _write_config_to_dest(self, destination, exports_dir, values=None):
         else:
             self._generate_default_oci_configuration(destination)
 
+    @staticmethod
+    def _get_manifest_attributes(manifest, attr_name, default_value):
+        """
+        Collects corresponding information from manifest file
+        based on attribtue name
+
+        :returns attribtues' corresponding value if it does exist
+        else returns a default value that is predefined
+        """
+        if manifest is not None and attr_name in manifest:
+            return manifest[attr_name]
+        return default_value
+
+    def _upgrade_tempfiles_clean(self, manifest, options, was_service_active):
+        """
+        When upgrading a container, we stop the service and remove previously
+        installed tmpfiles, before restarting the service
+
+        :param manifest: dictionary loaded from manifest json file
+        :param options: a dictionary which contains user input collectively
+        :param was_service_active: a boolean to indicate whether the container service was active
+        """
+        # The thought process here might be a bit tricky. When noContainerService is true, has_container_service
+        # will have the opposite value. Thus goes with the 'not' sign. Then because of the 'not', the default return
+        # value when noContainerService is not present should be False so that has_container_service will have
+        # a default value of True
+        has_container_service = not(SystemContainers._get_manifest_attributes(manifest, "noContainerService", False))
+
+        if has_container_service and options["upgrade_mode"] != SystemContainers.CHECKOUT_MODE_INSTALL:
+            if was_service_active:
+                self._systemctl_command("stop", options["name"])
+            if os.path.exists(options["tmpfilesout"]):
+                try:
+                    self._systemd_tmpfiles("--remove", options["tmpfilesout"])
+                except subprocess.CalledProcessError:
+                    pass
+
+    def _update_rename_file_value(self, manifest, values):
```

Comment:
Here, I think it might be better if we could just use 
```
util.write_template("manifest.json", rename_files, values, None)
```
right? A lot of code seems similar in this context. Any reasons for looping each single entries and do a replacement earlier? 

I might misunderstood parts here earlier, ignore this comment if I happened to be wrong =).

Reply:
I'll defer this to @giuseppe

Reply:
I think we cannot do it.  Here we iterate the `rename_files` map and replace the value of each file name, without writing it to a file.

We need this so we can rename installed files to the host, such as `"/etc/foo.conf" : "/etc/$NAME.conf"`.  This means the file `/etc/foo.conf` will be renamed before being installed to the host and its path will depend from the name of the container being installed.  We don't write its content, we only rename the file.

`util.write_template()` instead writes the content to a file.

Is this what you had in mind or have I misunderstood it?

Reply:
Thanks for the quick and meaningful response above =). 

> util.write_template() instead writes the content to a file.
Is this what you had in mind or have I misunderstood it?

I did have that in mind earlier (with destination parameter being None in the function call). But I was confused about `template.substitute` earlier, I accidentally thought it will do a substitution for dictionary as well. 

So yea, I think you are right, I think we can not adopt the implementation of `write_template` because `write_template` will not return new_value when destination = None

This is from write_template:
```
   if destination is not None:
        with open(destination, "w") as outfile:
            outfile.write(result)
        return result
    return None
```

Again, thanks for clarifying it  =)

Reply:
is the PR good for you or would you like to do something more?

I think it is ready to 🚢

Reply:
>is the PR good for you or would you like to do something more?

=), I think it is good for me. may be I will later do some follow up commits(unit tests for rpm_host_install), but I think that can be done in another pull request later. 

It should be ready to 🚢 if @ashcrow  is fine with me doing the unit test in a follow up pull request =P. 

P.S: thanks for the review!  and the late response was because I was grabbing lunch earlier.

Reply:
I'm fine with unittesting as a follow up ... just don't forget to do it 😄